numpy/core/tests/test_datetime.py

```diff
@@ -212,6 +211,18 @@ def test_datetime_scalar_construction(self):
         assert_equal(np.datetime64('1920-03-13T18:33:12.5', 'Y'),
                      np.datetime64('1920'))
 
+        # Construction from numpy types
+        assert_array_equal(np.datetime64(np.int32(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.int64(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.uint(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.uint32(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.uint64(1262304000), 's'),
```

Comment:
Ideally should test some value that won't fit in a `uintp`, like `np.uint64(2**48)` which will fail on 32-bit windows.